server/data/search.go

```diff
@@ -42,15 +42,14 @@ func SearchUsers(queryFields [][2]string) []*UserProfile {
 	rows, err := db.Query(context.Background(), sqlQuery, queryValues...)
 
 	if err != nil {
-		fmt.Println("Error searching for users in field. Error:", err.Error())
+		return []*UserProfile{}, err
```

Comment:
Maybe we should return a nil slice instead? Really there's no reason to allocate memory for a slice if the function returns an error.